### ISSUE TITLE
Incorrect signature given for method declared using (*args) instead of (self, *args)

### ISSUE THREAD

**Issue by anntzer** [ipython/ipython#8822]
```
In [1]: class C:
   ...:     def f(*args): pass
   ...:     def g(*args, **kwargs): pass
   ...: c = C()
   ...: 

In [2]: c.f?
Signature: c.f() # oops
Docstring: <no docstring>
File:      ~/<ipython-input-1-b2acea81d98f>
Type:      method

In [3]: c.g?
Signature: c.g(**kwargs) # oops
Docstring: <no docstring>
File:      ~/<ipython-input-1-b2acea81d98f>
Type:      method
```

One may argue that this is (very) poor style but at least the standard library's `inspect.Signature.bind` is declared like this.

**Comment by Carreau** [comment]
Marking as wishlist. If there is a fix we'll probably accept it.

**Comment by anntzer** [comment]
This was fixed in the stdlib at https://hg.python.org/cpython/rev/3544827d42e6
Given that `IPython.utils.signature` is only used for that purpose, would it be acceptable to try importing the stdlib's `inspect.signature` first (which will succeed for recent enough versions of Python) and then fallback on `IPython.util.signature` on failure?

**Comment by Carreau** [comment]
Seem reasonable. 
Do you want to make a PR ?

**Comment by anntzer** [comment]
Closed by #8823.
